pkg/roachprod/cloud/cluster_cloud.go

```diff
@@ -280,19 +280,13 @@ func CreateCluster(
 
 // DestroyCluster TODO(peter): document
 func DestroyCluster(l *logger.Logger, c *Cluster) error {
-	err := vm.FanOut(c.VMs, func(p vm.Provider, vms vm.List) error {
+	return vm.FanOut(c.VMs, func(p vm.Provider, vms vm.List) error {
 		// Enable a fast-path for providers that can destroy a cluster in one shot.
 		if x, ok := p.(vm.DeleteCluster); ok {
 			return x.DeleteCluster(l, c.Name)
 		}
 		return p.Delete(l, vms)
 	})
-	if err != nil {
-		return err
-	}
-	return vm.FanOutDNS(c.VMs, func(p vm.DNSProvider, vms vm.List) error {
-		return p.DeleteRecordsBySubdomain(c.Name)
```

Comment:
Do we _need_ to remove this? We can leave GC to remove dangling resources (such as when the roachtest process crashes).

Also, what happens to local clusters?